packages/flutter/lib/src/widgets/window.dart

```diff
@@ -2,5 +2,307 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// Placeholder to be used in a future version of Flutter.
-abstract final class Window {}
+import 'dart:ui' show AppExitType, FlutterView;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import '_window_ffi.dart' if (dart.library.js_util) '_window_web.dart' as window_impl;
+import 'binding.dart';
+import 'framework.dart';
+import 'view.dart';
+
+/// Defines the possible archetypes for a window.
+enum WindowArchetype {
+  /// Defines a traditional window
+  regular,
+}
+
+/// Defines the possible states that a window can be in.
+enum WindowState {
+  /// Window is in its normal state, neither maximized, nor minimized.
+  restored,
+
+  /// Window is maximized, occupying the full screen but still showing the system UI.
+  maximized,
+
+  /// Window is minimized and not visible on the screen.
+  minimized,
+}
```

Comment:
Bearing in mind what we talked about wrt. "rich" states, should we leave this out for now?

Or already go for:
```cpp
class WindowState {
}

class Restored : WindowState {
}

// ...
```